components/raftstore-v2/tests/integrations/test_read.rs

```diff
@@ -0,0 +1,213 @@
+    req.mut_header().set_peer(new_peer(1, 3));
+    req.mut_header().set_term(6);
+    req.mut_header().set_region_id(region_id);
+    req.mut_header()
+        .set_region_epoch(region.take_region_epoch());
+
+    for write_cmd in [
+        CmdType::Prewrite,
+        CmdType::Delete,
+        CmdType::DeleteRange,
+        CmdType::Put,
+        CmdType::IngestSst,
+    ] {
+        let mut request_inner = Request::default();
+        request_inner.set_cmd_type(write_cmd);
+        req.mut_requests().push(request_inner);
+        let res = router.query(region_id, req.clone()).unwrap();
+        let resp = res.read();
+        assert!(resp.is_none());
+        let error_resp = res.response().unwrap();
+        assert!(error_resp.get_header().has_error());
+    }
+}
+
+#[test]
+fn test_snap_with_invalid_parameter() {
+    let (_node, _transport, router) = super::setup_default_cluster();
+    std::thread::sleep(std::time::Duration::from_millis(200));
+    let region_id = 2;
+    let mut req = RaftCmdRequest::default();
+    req.mut_header().set_peer(new_peer(1, 3));
+    req.mut_status_request()
+        .set_cmd_type(StatusCmdType::RegionDetail);
+    let res = router.query(region_id, req.clone()).unwrap();
+    let status_resp = res.response().unwrap().get_status_response();
+    let detail = status_resp.get_region_detail();
+    let mut region = detail.get_region().clone();
+    let mut region_epoch = region.take_region_epoch();
+
+    let mut req = RaftCmdRequest::default();
+    // store_id is incorrect;
+    req.mut_header().set_peer(new_peer(2, 3));
+    req.mut_header().set_term(6);
+    req.mut_header().set_region_id(region_id);
+    req.mut_header().set_region_epoch(region_epoch.clone());
+    let mut request_inner = Request::default();
+    request_inner.set_cmd_type(CmdType::Snap);
+    req.mut_requests().push(request_inner);
+    let res = router.query(region_id, req.clone()).unwrap();
+    let error_resp = res.response().unwrap();
+    assert!(error_resp.get_header().has_error());
+
+    // run again, with incorrect peer_id
+    req.mut_header().set_peer(new_peer(1, 4));
+    let res = router.query(region_id, req.clone()).unwrap();
+    let error_resp = res.response().unwrap();
+    assert!(error_resp.get_header().has_error());
+
+    // run with stale term
+    req.mut_header().set_term(1);
+    req.mut_header().set_peer(new_peer(1, 3));
+    let res = router.query(region_id, req.clone()).unwrap();
+    let error_resp = res.response().unwrap();
+    assert!(error_resp.get_header().has_error());
+
+    // run again with invalid region_epoch
+    region_epoch.set_version(region_epoch.get_version() + 1);
+    req.mut_header().set_region_epoch(region_epoch.clone());
+    req.mut_header().set_term(6);
+    let res = router.query(region_id, req.clone()).unwrap();
+    let error_resp = res.response().unwrap();
+    assert!(error_resp.get_header().has_error());
+}
+
+#[test]
+fn test_read_stale() {
```

Comment:
Why not merge it with test_snap_with_invalid_parameter?

Reply:
good point.